Sources/Support/StoreMessagesHelper.swift

```diff
@@ -0,0 +1,195 @@
+               types.contains(messageType) {
+                    do {
+                        try message.display(in: self.systemInfo.currentWindowScene)
+                    } catch {
+                        Logger.error("Error displaying StoreKit message: \(error)")
+                    }
+            }
+        }
+    }
+
+    #endif
+}
+
+// @unchecked because:
+// - Class is not `final` (it's mocked). This implicitly makes subclasses `Sendable` even if they're not thread-safe.
+// - It has mutable `_deferredMessages` which is necessary due to the availability annotations.
+extension StoreMessagesHelper: @unchecked Sendable {}
+
+protocol StoreMessagesProvider {
+
+    #if os(iOS) || targetEnvironment(macCatalyst) || VISION_OS
+
+    @available(iOS 16.0, *)
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    var messages: any StoreMessageAsyncSequence { get }
+
+    #endif
+}
+
+protocol StoreMessage: Sendable {
+
+    #if os(iOS) || targetEnvironment(macCatalyst) || VISION_OS
+
+    @available(iOS 16.0, *)
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    var reason: Message.Reason { get }
+
+    @available(iOS 16.0, *)
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    @MainActor func display(in scene: UIWindowScene) throws
+
+    #endif
+}
+
+#if os(iOS) || targetEnvironment(macCatalyst) || VISION_OS
+
+@available(iOS 13.0, *)
+@rethrows protocol StoreMessageAsyncIteratorProtocol: AsyncIteratorProtocol where Element == StoreMessage { }
+
+@available(iOS 13.0, *)
+@rethrows protocol StoreMessageAsyncSequence: AsyncSequence where AsyncIterator: StoreMessageAsyncIteratorProtocol { }
+
+@available(iOS 16.0, *)
+@available(macOS, unavailable)
+@available(watchOS, unavailable)
+@available(tvOS, unavailable)
+private final class StoreMessageWrapper: StoreMessage {
+
+    private let message: Message
+
+    init(message: Message) {
+        self.message = message
+    }
+
+    var reason: Message.Reason {
+        return self.message.reason
+    }
+
+    func display(in scene: UIWindowScene) throws {
+        try self.message.display(in: scene)
+    }
+}
+
+@available(iOS 16.0, *)
+@available(macOS, unavailable)
+@available(watchOS, unavailable)
+@available(tvOS, unavailable)
+private struct StoreMessageSequence: StoreMessageAsyncSequence {
+
+    struct AsyncIterator: StoreMessageAsyncIteratorProtocol {
+
+        var iterator: StoreKit.Message.Messages.AsyncIterator
+
+        mutating func next() async -> StoreMessage? {
+            await iterator.next().map {
+                StoreMessageWrapper(message: $0)
+            }
+        }
+
+    }
+
+    let underlyingSequence: Message.Messages
+
+    func makeAsyncIterator() -> AsyncIterator {
+        AsyncIterator(iterator: underlyingSequence.makeAsyncIterator())
+    }
+
+}
```

Comment:
I think there's an alternative way that doesn't require this manual creation of a sequence. I'll ping you in Slack to help you with that.